### PR TITLE
Don't return a traceback if the password is wrong.

### PR DESCRIPTION
**Pull request by juanjo-nan**
Instead return a UserError.

Task #160407

### CODE REVIEW

File: invoice.py [hunk 1]
```diff
@@ -454,12 +454,15 @@ def _sign_facturae(self, xml_string, certificate_password):
...
-
+            try:
+                (
+                    private_key,
+                    certificate,
+                    additional_certificates,
+                    ) = pkcs12.load_key_and_certificates(cert, password)
+            except ValueError:
+                raise UserError(gettext(
+                    'account_invoice_facturae.msg_incorrect_password'))
```
Comment (raimonesteve) [inline invoice.py:465]: At the moment, except return:

```
/cryptography/hazmat/backends/openssl/backend.py
Invalid password or PKCS12 data
```

but the error could it from other reasons besides the password.

I thing we need to capture the error message and not a fixed message.

Comment (juanjo-nan) [reply invoice.py]: Changed to the error message: https://github.com/NaN-tic/trytond-account_invoice_facturae/pull/7/commits/42067cbca36e90e8c825b4cbffa0a875ee7613e0

File: invoice.py [hunk 2]
```diff
@@ -454,12 +454,16 @@ def _sign_facturae(self, xml_string, certificate_password):
...
+            try:
+                (
+                    private_key,
+                    certificate,
+                    additional_certificates,
+                    ) = pkcs12.load_key_and_certificates(cert, password)
+            except ValueError as e:
+                raise UserError(gettext(
+                    'account_invoice_facturae.msg_certificate_error',
+                    error=e))
```
Comment (raimonesteve) [inline invoice.py:466]: Ho convertiria a string: 

```
error=str(e)
```

Comment (albertca) [reply invoice.py:466]: Estic d'acord amb el Raimon

**Review by albertca — APPROVED** [review]

**Pull request merged**